IMDb.js

```diff
@@ -36,100 +35,114 @@
+	var found = false;
+	var rows = ZU.xpath(doc, '//td[contains(@class, "result_text")]');
+	for (var i=0; i<rows.length; i++) {
+		var href = ZU.xpathText(rows[i], './a/@href');
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
 
-		Zotero.selectItems(items, function (items) {
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
-				Zotero.done();
 				return true;
 			}
+			var articles = [];
 			for (var i in items) {
-				ids.push(i);
+				articles.push(i);
 			}
-			apiFetch(ids);
+			ZU.processDocuments(articles, scrape);
 		});
 	} else {
-		var id = url.match(/\/title\/(tt\d+)/)[1];
-		apiFetch([id]);
+		scrape(doc, url);
 	}
 }
 
-// Takes IMDB IDs and makes items
-function apiFetch(ids) {
-	var apiRoot = "http://omdbapi.com/?i="; //&tomatoes=true removed as it often breaks the API
-	for (i in ids) ids[i] = apiRoot + ids[i];
-	Zotero.Utilities.doGet(ids, parseIMDBapi);
-}
 
-// parse result from imdbapi.com
-// should be json
-function parseIMDBapi(text, response, url) {
-	//Zotero.debug(url);
-	//Zotero.debug(text);
-	try {
-		var obj = JSON.parse(text);
-	} catch (e) {
-		Zotero.debug("JSON parse error");
-		throw e;
-	}
-	var item = new Zotero.Item("film");
-	item.title = obj.Title;
-	item.date = obj.Released ? obj.Released : obj.Year;
-	item.genre = obj.Genre;
-	if (obj.Director) item = addCreator(item, obj.Director, "director");
-	if (obj.Writer) item = addCreator(item, obj.Writer, "scriptwriter");
-	if (obj.Actors) item = addCreator(item, obj.Actors, "contributor");
-	item.abstractNote = obj.Plot;
-	item.attachments.push({
-		url: obj.Poster,
-		title: "Poster"
+function scrape(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	//translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		var titleWrapper = ZU.xpath(doc, '//div[contains(@class, "title_wrapper")]');
+		var title = ZU.xpathText(titleWrapper, './h1/text()[1]');
+		if (title) {
+			item.title = title;
+		}
+		item.date = ZU.xpathText(titleWrapper, './/meta[@itemprop="datePublished"]/@content');
+		item.runningTime = ZU.xpathText(titleWrapper, './/time[@itemprop="duration"]');
+		item.genre = ZU.xpathText(titleWrapper, './/span[@itemprop="genre"]');
+		var origTitle = ZU.xpathText(titleWrapper, './/div[contains(@class, "originalTitle")]/text()[1]');
+		if (origTitle) {
+			addExtra(item, "original-title: "+origTitle);
+		}
+		var pageId = ZU.xpathText(doc, '//meta[@property="pageId"]/@content');
+		if (pageId) {
+			addExtra(item, "IMDb ID: "+pageId);
+		}
+		
+		var summary = ZU.xpath(doc, '//div[contains(@class, "plot_summary_wrapper")]');
+		var creatorsMapping = {
```

Comment:
I haven't seen any other role in the short list in the overview e.g. http://www.imdb.com/title/tt0089276/ but there are much more persons in the long list e.g. http://www.imdb.com/title/tt0089276/fullcredits?ref_=tt_ov_st_sm . Do you think it is worth to grab more persons from there? Zotero has AFAIK only another role `producer` for `film`s. But I wasn't even sure about the many actors...